cli/src/main/java/com/devonfw/tools/ide/merge/xmlmerger/XmlMergeSupport.java

```diff
@@ -0,0 +1,332 @@
+package com.devonfw.tools.ide.merge.xmlmerger;
+
+import java.io.StringWriter;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.function.Predicate;
+import javax.xml.XMLConstants;
+import javax.xml.namespace.QName;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Attr;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+/**
+ * Utility class with static helper methods for XML merging.
+ */
+public interface XmlMergeSupport {
+
+  /**
+   * The XML namespace used by this {@link XmlMerger} to configure merge specific meta-information.
+   *
+   * @see XmlMergeSupport#hasMergeNamespace(Attr)
+   */
+  String MERGE_NS_URI = "https://github.com/devonfw/IDEasy/merge";
+
+  /** {@link Attr#getName() Attribute name} {@value} */
+  String ATTR_ID = "id";
+
+  /** XPath for {@link Attr} {@value} */
+  String XPATH_ATTR_ID = "@id";
+
+  /** {@link Attr#getName() Attribute name} {@value} */
+  String ATTR_NAME = "name";
+
+  /** XPath for {@link Attr} {@value} */
+  String XPATH_ATTR_NAME = "@name";
+
+  /** XPath for {@link Element#getTagName() element tag name}: {@value} */
+  String XPATH_ELEMENT_NAME = "name()";
+
+  /** {@link Attr#getName() Attribute name} {@value} */
+  String ATTR_STRATEGY = "strategy";
+
+  /** XPath for {@link Element#getTextContent() element text}: {@value} */
+  String XPATH_ELEMENT_TEXT = "text()";
+
+  /**
+   * @param attribute the XML {@link Attr} to check.
+   * @return {@code true} if the {@code attribute} is specific for our XML merger and related to {@link XmlMergeSupport#MERGE_NS_URI} (that needs to be omitted
+   *     in the XML output), {@code false} otherwise (regular {@code attribute} for the target payload).
+   */
+  static boolean hasMergeNamespace(Attr attribute) {
+
+    if (MERGE_NS_URI.equals(attribute.getNamespaceURI())) {
+      return true;
+    } else {
+      return "xmlns".equals(attribute.getPrefix()) && MERGE_NS_URI.equals(attribute.getValue());
+    }
+  }
+
+  /**
+   * @param element the {@link Element} for which the XPath is requested.
+   * @return the XPath expression.
+   */
+  static String getXPath(Element element) {
+
+    return getXPath(element, false);
+  }
+
+  /**
+   * @param element the {@link Element} for which the XPath is requested.
+   * @param includeAttributes {@code true} to also include the attributes of
```

Comment:
Javadoc not complete?

Reply:
Thanks for your finding - seems I was in the middle of writing this and then got distracted. I fixed this.